webpack.config.js

```diff
@@ -102,11 +102,43 @@ const webConfig = {
     entry: {
         'src/extensionWeb': './src/extensionWeb.ts',
     },
-    plugins: baseConfig.plugins?.concat(
+    plugins: (baseConfig.plugins ?? []).concat(
         new webpack.optimize.LimitChunkCountPlugin({
             maxChunks: 1, // disable chunks by default since web extensions must be a single bundle
+        }),
+        // some modules are provided globally and dont require an explicit import,
+        // so we need to polyfill them using this method
+        new webpack.ProvidePlugin({
+            process: require.resolve('process/browser'),
+            Buffer: ['buffer', 'Buffer'],
+        }),
+        new webpack.EnvironmentPlugin({
+            NODE_DEBUG: 'development',
+            READABLE_STREAM: 'disable',
         })
     ),
+    resolve: {
+        extensions: ['.ts', '.js'],
+        fallback: {
+            fs: require.resolve('browserify-fs'),
+            assert: require.resolve('assert/'),
+            constants: require.resolve('constants-browserify'),
+            crypto: require.resolve('crypto-browserify'),
+            http: require.resolve('stream-http'),
+            https: require.resolve('https-browserify'),
+            os: require.resolve('os-browserify/browser'),
+            path: require.resolve('path-browserify'),
+            stream: require.resolve('stream-browserify'),
+            zlib: require.resolve('browserify-zlib'),
+            process: require.resolve('process/browser'),
+            // TODO: these do not have a straight-forward replacement and cause browser mode not to work
+            child_process: false, // Reason for error: 'TypeError: The "original" argument must be of type Function'
+            async_hooks: false,
```

Comment:
It's only used for telemetry, but yeah a replacement is very difficult to do without access to the engine's APIs. There's a partial solution in `src/shared/telemetry/spans.ts` that could be used. The behavior only diverges when two separate async tasks are trying to record telemetry to the context in parallel. CodeWhisperer doesn't do this though.